forum/communications/views.py

```diff
@@ -1,3 +1,41 @@
 from django.shortcuts import render
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
+from .models import Message
+from .serializers import ChatRoomSerializer, MessageSerializer
+from django.contrib.auth import get_user_model
 
-# Create your views here.
+User = get_user_model()
+
+
+def index(request):
+    return render(request, "communications/index.html")
+
+
+def room(request, room_name):
+    return render(request, "communications/room.html", {"room_name": room_name})
+
+
+class CreateConversationView(generics.CreateAPIView):
+    serializer_class = ChatRoomSerializer
+    permission_classes = [IsAuthenticated]
+
+    def perform_create(self, serializer):
+        serializer.save()
```

Comment:
Currently, no checks ensure the participants are valid users or not already in a conversation.
Add validation for unique conversations and valid participants.

_Example:_

```
from rest_framework.exceptions import ValidationError

def perform_create(self, serializer):
    participants = serializer.validated_data.get("participants")
    if len(participants) != 2:
        raise ValidationError("A conversation must have exactly two participants.")
    existing_room = Room.objects.filter(name=f"room_{participants[0].id}_{participants[1].id}").exists()
    if existing_room:
        raise ValidationError("This conversation already exists.")
    serializer.save()
```

Reply:
Done